### PR TITLE
Add a wrapper for strings containing HTML.

### LINKED ISSUE #182: URL formatting goes wild
There is an issue in the markup of URLs resulting from a `get` command. The <a> tag conflicts with the *highlighting* feature (which is supposed to highlight the part of the URL that matches the query; resulting in an ugly html tag leakage.

### PR DESCRIPTION
**Pull request by hmil**
This layer catches most XSS and double escapes.
Multiple XSS are fixed in this patch.
Fixes #182, caused by an HTML tag leakage.

### CODE REVIEW

File: packages/vaultage-ui-webcli/src/VaultEntryFormatter.ts [hunk 1]
```diff
@@ -8,98 +9,49 @@ export class VaultEntryFormatter {
...
-        e.url = this.highlight(e.url, highlights);
-
-        return this.formatSingle(e);
+    public static formatSingle(e: IVaultDBEntry): SanitizedString {
+
+        const extraClass =  (e.password_strength_indication === PasswordStrength.WEAK) ? 'weakPassword' :
+                            (e.password_strength_indication === PasswordStrength.MEDIUM) ? 'mediumPassword' :
+                            '';
+
+        return html`<span class="entry">
```
Comment (hmil) [inline packages/vaultage-ui-webcli/src/VaultEntryFormatter.ts]: Introducing: The holy grail of HTML escaping. ES6 tagged templates!
(All of the shit below is escaped **and** checked against double-escape.

File: packages/vaultage-ui-webcli/src/commands/Get.ts [hunk 2]
```diff
@@ -16,13 +17,13 @@ export class GetCommand implements ICommand {
...
         if (searchTerms.length === 0) {
-            this.shell.echoHTML('Usage: get <i>term1</i> <i>[term2]</i>... (use \'ls\' to list all entries).');
+            this.shell.echoHTML(html`Usage: get <i>term1</i> <i>[term2]</i>... (use \'ls\' to list all entries).`);
             return;
         }
 
         const results = this.ctx.vault.findEntries(...searchTerms);
         const coloredSearchString = VaultEntryFormatter.searchTermsToHighlightedString(searchTerms);
-        this.shell.echoHTML('Searching for ' + coloredSearchString + ', ' + results.length + ' matching entries.');
+        this.shell.echoHTML(html`Searching for ${coloredSearchString}, ${results.length} matching entries.`);
```
Comment (hmil) [inline packages/vaultage-ui-webcli/src/commands/Get.ts]: This kind of XSS is hard to spot, but with this patch, we can rely on TypeScript to check it for us.

File: packages/vaultage-ui-webcli/src/webshell/Shell.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export class Shell implements ICommandHandler {
     /**
      * Main text shown when clearing the screen
      */
-    private bannerHTML: string = 'Welcome to my shell!';
+    private bannerHTML: SanitizedString = html`Welcome to my shell!`;
```
Comment (hmil) [inline packages/vaultage-ui-webcli/src/webshell/Shell.ts]: You never want to use the SanitizedString constructor. Use the template string form instead.

File: packages/vaultage-ui-webcli/src/security/xss.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    public transform(transformation: (e: string) => string): SanitizedString {
+        return new SanitizedString(transformation(this.valueOf()));
+    }
+}
+
+/**
+ * Escapes html tags and attributes from the string, making it safe.
+ */
+export function escape(input: string): SanitizedString  {
```
Comment (lbarman) [inline packages/vaultage-ui-webcli/src/security/xss.ts]: Where does this function come from ? Is it standard ?

Comment (hmil) [reply packages/vaultage-ui-webcli/src/security/xss.ts]: Nope, there is no standard way of escaping HTML in JS... (After all, it's just the one language that interacts with HTML the most).
I found a good example in mustache.js though, which handles additional cases. I'll update this pr.
https://github.com/janl/mustache.js/commit/378bcca8a5cfe4058f294a3dbb78e8755e8e0da5

Comment (lbarman) [reply packages/vaultage-ui-webcli/src/security/xss.ts]: OK, thanks. R4M in my opinion

**Pull request merged**